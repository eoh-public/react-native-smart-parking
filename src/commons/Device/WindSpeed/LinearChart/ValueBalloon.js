import React, { memo } from 'react';
import { G, Rect } from 'react-native-svg';

const ValueBalloon = memo((props) => {
  const { x, y } = props;
  const transform = `translate(${x - 24},${y - 54})`;

  return (
    <G transform={transform}>
      <Rect
        x={23.5752}
        y={29.3477}
        width={9.03012}
        height={9.03012}
        rx={1.12876}
        transform="rotate(45 23.5752 29.3477)"
        fill="#262626"
      />
      <Rect width={47.1506} height={36.1205} rx={3.38629} fill="#262626" />
    </G>
  );
});

export default ValueBalloon;
