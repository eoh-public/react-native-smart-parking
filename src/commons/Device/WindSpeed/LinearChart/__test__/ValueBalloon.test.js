import ValueBalloon from '../ValueBalloon';
import React from 'react';
import renderer, { act } from 'react-test-renderer';

describe('Test ValueBalloon', () => {
  let wrapper;

  test('ValueBalloon render', () => {
    act(() => {
      wrapper = renderer.create(<ValueBalloon x={10} y={15} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
