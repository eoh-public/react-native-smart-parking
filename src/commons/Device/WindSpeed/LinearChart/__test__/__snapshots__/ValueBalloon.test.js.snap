// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ValueBalloon ValueBalloon render 1`] = `
<RNSVGGroup
  matrix={
    Array [
      1,
      0,
      0,
      1,
      -14,
      -39,
    ]
  }
>
  <RNSVGRect
    fill={4280690214}
    height={9.03012}
    matrix={
      Array [
        0.7071067811865476,
        0.7071067811865475,
        -0.7071067811865475,
        0.7071067811865476,
        27.656973894399343,
        -8.07444147005754,
      ]
    }
    propList={
      Array [
        "fill",
      ]
    }
    rx={1.12876}
    width={9.03012}
    x={23.5752}
    y={29.3477}
  />
  <RNSVGRect
    fill={4280690214}
    height={36.1205}
    propList={
      Array [
        "fill",
      ]
    }
    rx={3.38629}
    width={47.1506}
    x={0}
    y={0}
  />
</RNSVGGroup>
`;
